h/claim/test/views_test.py

```diff
@@ -0,0 +1,208 @@
+    config.registry.registerUtility(Mock(), IResetPasswordForm)
+    config.registry.registerUtility(Mock(), IResetPasswordSchema)
+
+
+def _claim_account_request():
+    request = DummyRequest()
+    request.user = None
+    request.route_url = Mock(return_value='<url>')
+
+    token = request.registry.claim_serializer.dumps({'userid': 1})
+    request.matchdict['token'] = token
+
+    return request
+
+
+def _update_account_request(post=None):
+    post = post or {'password': 'safeashouses'}
+    request = DummyRequest(post=post)
+    request.user = None
+    request.route_url = Mock(return_value='<url>')
+
+    token = request.registry.claim_serializer.dumps({'userid': 1})
+    request.matchdict['token'] = token
+    return request
+
+
+def _fake_user():
+    return User()
+
+
+def test_claim_account_returns_form(config):
+    configure(config)
+    request = _claim_account_request()
+
+    with patch('h.accounts.models.User.get_by_id') as mock_get_by_id:
+        mock_get_by_id.return_value = _fake_user()
+
+        from ..views import claim_account
+        ctx = claim_account(request)
+        assert 'form' in ctx
+
+
+def test_claim_account_when_signed_in(config):
+    configure(config)
+    request = _claim_account_request()
+    request.user = 'billy'
+
+    from ..views import claim_account
+
+    with raises(HTTPFound) as execinfo:
+        claim_account(request)
+
+    assert execinfo.value.location == '<url>'
+    assert request.route_url.called_with('stream')
+
+
+def test_claim_account_with_invalid_token(config):
+    configure(config)
+    request = _claim_account_request()
+    request.matchdict['token'] = 'foobar'
+
+    from ..views import claim_account
+
+    with raises(ValueError) as execinfo:
+        claim_account(request)
+
+    assert str(execinfo.value) == 'Invalid signature'
+
+
+def test_claim_account_when_already_registered(config):
+    configure(config)
+    request = _claim_account_request()
+
+    with patch('h.accounts.models.User.get_by_id') as mock_get_by_id:
+        user = _fake_user()
+        user.password = '<password>'
+        mock_get_by_id.return_value = user
+
+        from ..views import claim_account
+        with raises(HTTPFound) as execinfo:
+            claim_account(request)
+
+    assert execinfo.value.location == '<url>'
+    request.route_url.assert_called_with('stream')
+
+
+def xtest_claim_account_already_registered_message(config):
+    configure(config)
+    request = _claim_account_request()
+
+    with patch('h.accounts.models.User.get_by_id') as mock_get_by_id:
+        with patch('horus.lib.FlashMessage') as mock_flash:
+            user = _fake_user()
+            user.password = '<password>'
+            mock_get_by_id.return_value = user
+
+            from ..views import claim_account
+            with raises(HTTPFound):
+                claim_account(request)
+
+    msg = 'This account has already been claimed'
+    mock_flash.assert_called_with(request, msg, kind='error')
```

Comment:
Could not for the life of me get this patch to work. I've no idea what I'm doing wrong, but the claim_account method runs with the original FlashMessage function.